WorkspaceServer.Tests/Kernel/LanguageKernelTests.cs

```diff
@@ -45,6 +48,141 @@ public async Task it_returns_the_result_of_a_non_null_expression()
                 .Be(123);
         }
 
+        // Option 1: inline switch
+        [Theory]
+        [InlineData(Language.FSharp)]
+        [InlineData(Language.CSharp)]
+        public async Task it_remembers_state_between_submissions(Language language)
+        {
+            string[] source = language switch
+            {
+                Language.FSharp => new[]
+                {
+                    "let add x y = x + y",
+                    "add 2 3"
+                },
+
+                Language.CSharp => new[]
+                {
+                    "int Add(int x, int y) { return x + y; }",
+                    "Add(2, 3)"
+                },
+                _ => throw new InvalidOperationException("Un supported Language for this test case")
+            };
+
+            var kernel = CreateKernel(language);
+
+            await SubmitSource(kernel, source);
+
+            KernelEvents.ValuesOnly()
+                .OfType<ReturnValueProduced>()
+                .Last()
+                .Value
+                .Should()
+                .Be(5);
+        }
+
+
+        // Option 2:  Use InlineData
+        [Theory]
+        [InlineData(Language.FSharp, "let add x y = x + y", 
+                                     "add 2 3")]
+        [InlineData(Language.CSharp, "int Add(int x, int y) { return x + y; }",
+                                     "Add(2, 3)")]
+        public async Task it_remembers_state_between_submissions_option2(Language language, string line1, string line2)
+        {
+            var kernel = CreateKernel(language);
+
+            await SubmitSource(kernel, line1);
+            await SubmitSource(kernel, line2);
+
+            KernelEvents.ValuesOnly()
+                .OfType<ReturnValueProduced>()
+                .Last()
+                .Value
+                .Should()
+                .Be(5);
+        }
+
+        public T GetElementForLanguage<T>(Language language, params (Language lang, T value)[] values)
+        {
+            // The use of `Single()` ensures that the test will fail if a value was not given for a specific language.
+            return values.Single(val => val.lang == language).value;
+        }
+        
+        //Option 3: GetElementForLanguage iter 1
+        [Theory]
+        [InlineData(Language.FSharp)]
+        [InlineData(Language.CSharp)]
+        public async Task it_remembers_state_between_submissions_option3(Language language)
```

Comment:
I'm going to cast my vote for this style of test case because:

1.  The code to select different values is short and easy to read.  I find the `switch` expression difficult because it's so verbose and requires a `default` case that must explicitly throw, while the helper method here handles that for you.
1.  Placing code in `[InlineData]` works for compile-time constants, but won't be usable for expressions.  We don't have a test that does this currently, but I can easily see a test that needs to call `GetElementForLanguage()` where the item returned is a `Type` or `Exception` or something that can't be placed in an attribute.
1.  Finally, I like that the code variants are declared exactly where and when they're used instead of at the top of the function where I have to trace a variable through.